monai/losses/focal_loss.py

```
@@ -156,10 +164,11 @@ def forward(self, input: torch.Tensor, target: torch.Tensor) -> torch.Tensor:
         loss: Optional[torch.Tensor] = None
         input = input.float()
         target = target.float()
+
         if self.use_softmax:
             if not self.include_background and self.alpha is not None:
-                self.alpha = None
-                warnings.warn("`include_background=False`, `alpha` ignored when using softmax.")
+                if isinstance(self.alpha, (float, int)):
+                    warnings.warn("`include_background=False`, scalar `alpha` ignored when using softmax.")
```

Ruff (0.14.10)

No explicit `stacklevel` keyword argument found

Set `stacklevel=2`

(B028)